src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -212,19 +214,21 @@ static ScriptInfo removeTriggers(File script) {
 				i--;
 			}
 		}
-		
+
 		info.commands = Commands.unregisterCommands(script);
-		
+
 		return info;
 	}
-	
+
 	static void removeAllTriggers() {
 		triggers.clear();
 		for (Trigger t : selfRegisteredTriggers)
 			((SelfRegisteringSkriptEvent) t.getEvent()).unregisterAll();
 		selfRegisteredTriggers.clear();
 	}
-	
+
+	private static final Set<HandlerList> handlerListsRegisteredTo  = new HashSet<>();
```

Comment:
This should be moved to the PriorityListener class, because events are registered per priority